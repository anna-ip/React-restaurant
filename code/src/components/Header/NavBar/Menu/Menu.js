import React from 'react'
import styles from './Menu.module.scss'
import Links from '../../../Links/Links'

function Menu({ open, setOpen}) {

  return (
    <div>
      <nav
        open={open}
        className={!open ? [styles.nav] : [styles.navOpen]}
      >
        <ul className={!open ? [styles.menuNav] : [styles.menuNavOpen]}>
          <li className={styles.menuNavItem}>
            <Links to={'/'} text={'Home'} >
              Home
            </Links>
            <Links to={'/menu'} text={'Menu'}>
              Menu
            </Links>
            <Links to={'/'} text={'About'}>
              About
            </Links>
            <Links to={'/'} text={'Contact'}>
              Contact
            </Links>
          </li>
        </ul>
      </nav>
    </div>
  )
}

export default Menu
