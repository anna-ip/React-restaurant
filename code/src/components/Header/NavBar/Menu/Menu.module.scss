@import '../../../../styles/variables';
@import '../../../../styles/mixins';

.nav {
  display: flex;
  flex-direction: row;
  justify-content: center;

  @media (max-width: $layout-breakpoint-small) {
    visibility: hidden;
  }
}

.menuNavItem {
  list-style-type: none;
}

.navOpen {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  opacity: 0.95;
  visibility: visible;
  z-index: 2;

  .menuNav {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
    overflow: hidden;
    background: $dark-color;
    list-style-type: none;
    padding-right: 1rem;
    transform: translateY(-100%);
    @include transition-ease;
  }

  .menuNavOpen {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
    overflow: hidden;
    background: $dark-color;
    list-style-type: none;
    padding-right: 1rem;
    margin: 0;
    transform: translateY(0);

    .menuNavItem {
      display: flex;
      flex-direction: column;
    }
  }
}
