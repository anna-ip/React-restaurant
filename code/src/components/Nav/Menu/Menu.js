import React from "react";
import useViewPort from "../../../hooks/useViewport";
import styles from "./Menu.module.scss";
import NavButtons from "./Links/NavButtons";

function Menu({ open, setOpen }) {
  const { width } = useViewPort();
  const breakpoint = 680;

  const handleRedirect = () => {
    width < breakpoint ? setOpen(!open) : setOpen(open);
  };

  return (
    <>
      <nav open={open} className={!open ? [styles.nav] : [styles.navOpen]}>
        <ul className={!open ? [styles.menuNav] : [styles.menuNavOpen]}>
          <li>
            <div
              className={styles.menuNavItem}
              onClick={() => handleRedirect()}
              onKeyPress={handleRedirect}
              role="presentation"
            >
              <NavButtons />
            </div>
          </li>
        </ul>
      </nav>
    </>
  );
}

export default Menu;
