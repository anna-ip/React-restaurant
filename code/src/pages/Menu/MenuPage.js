import React from "react";
import { MenuContext } from "../../App";
import FilterBtn from "../../components/MenuPageComponents/FilterBtn/FilterBtn";
import MenuCard from "../../components/MenuPageComponents/MenuCard/MenuCard";
import styles from "./MenuPage.module.scss";

const MenuPage = () => {
  const [storeMenus] = React.useContext(MenuContext);
  const menus = storeMenus.menus;
  const [filteredMenu, setFilteredMenu] = React.useState(menus || []);

  const filterMenuClick = (e) => {
    const btnValue = e.target.value;

    if (btnValue === "all") {
      setFilteredMenu(menus);
    } else {
      const filteredMenuItems = filterMenus(menus, btnValue);
      setFilteredMenu(filteredMenuItems);
    }
  };

  return (
    <div className={styles.container}>
      <div className={styles.btnContainer}>
        <FilterBtn filterMenuClick={filterMenuClick} />
      </div>
      <section className={styles.cardContainer}>
        <MenuCard filteredMenu={filteredMenu} />
      </section>
    </div>
  );
};

export const filterMenus = (menus, val) => {
  return menus.filter((menu) => menu.category === val);
};

export default MenuPage;
