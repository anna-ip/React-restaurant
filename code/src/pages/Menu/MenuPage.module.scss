@import '../../styles/_variables';
@import '../../styles/mixins';

.container {
  @include flexCenter;
}
.btnContainer {
  @include flex(row);
  justify-content: space-evenly;
  width: 100%;
  border-top: 1px solid $red-color;
  border-bottom: 1px solid $red-color;
  padding: 0.5rem 0;
}
.cardContainer {
  @include flex(column);
  justify-content: center;
  margin: 1.5rem 0;

  @media (min-width: $layout-breakpoint-small) {
    flex-direction: row;
    flex-wrap: wrap;
  }
}
