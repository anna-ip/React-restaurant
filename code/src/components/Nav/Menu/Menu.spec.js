import React from 'react'
import { shallow } from 'enzyme'
import Menu from './Menu'

describe('<Menu/>', () => {
  test('Menu render correctly', () => {
    const wrapper = shallow(<Menu />)
    expect(wrapper).toMatchSnapshot()
  })
})
