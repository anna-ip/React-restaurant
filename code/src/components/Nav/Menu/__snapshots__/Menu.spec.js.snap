// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> Menu render correctly 1`] = `
<Fragment>
  <nav
    className={
      Array [
        "nav",
      ]
    }
  >
    <ul
      className={
        Array [
          "menuNav",
        ]
      }
    >
      <li>
        <div
          className="menuNavItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="presentation"
        >
          <NavButtons />
        </div>
      </li>
    </ul>
  </nav>
</Fragment>
`;
